msyncd/synchronizer.cpp

```diff
@@ -2044,13 +2045,14 @@ void Synchronizer::removeExternalSyncStatus(const SyncProfile *aProfile)
 {
     int accountId = aProfile->key(KEY_ACCOUNT_ID).toInt();
     if (accountId) {
-        if (iExternalSyncProfileStatus.contains(accountId)) {
+        const QString &profileName = aProfile->name();
+        if (iExternalSyncProfileStatus.contains(profileName)) {
             // if profile was set to sync externally emit the change state signal
-            if (iExternalSyncProfileStatus.value(accountId)) {
+            if (iExternalSyncProfileStatus.value(profileName)) {
                 emit syncedExternallyStatus(accountId, aProfile->clientProfile()->name(),false);
```

Comment:
will aProfile->clientProfile()->name() differ from aProfile->name()?  I guess if the aProfile is built from a storage profile and a client profile subprofile, or something?  But if so, do we want to emit with the subprofile name or the top-level name here anyway?